### DISCUSSION TITLE
Whats the new import statement for role_grants?

### DISCUSSION THREAD

**Discussion by jayjay-ayjay** [Snowflake-Labs/terraform-provider-snowflake#3179] [Q&A] [answered]
I want to migrate my roles and role grants from Provider v0.71 -> v0.92 with "No-Downtime" for specific resources. Thus, I dumped the deprecated resources, recreated them with the new rules and deleted the old implementations of role_grants out of the state file. 
Though, I cant figure out how to import the new role_grants into the terraform state. 
I've tried this: 

`terraform import module.SNOWFLAKE_ROLES[\"MY_PARENT_ROLE\"].snowflake_grant_account_role.role_grants "MY_CHILD_ROLE|MY_PARENT_ROLE|"`

Throwing this error: 
Error: Configuration for import target does not exist
I have the according resource in Snowflake and it's configurated in my terraform code as well. 

Can anybody help?
Thanks in Advance!

**Comment by sfc-gh-asawicki** [comment]
Hey @jayjay-ayjay. Thanks for reaching out to us.

The import statement is documented here: https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/grant_account_role#import.

**Reply by jayjay-ayjay** [reply]
Thanks for your reply Artur! Should've mentioned that I've tried this as well. 
It throughs this error: "The import command expects two arguments."

**Reply by sfc-gh-asawicki** [reply]
This is the terraform CLI error, not the provider one. What's the command you run following the docs I referenced?

**Reply by jayjay-ayjay** [reply]
I ran this command:
terraform import "\"MY_CHILD_ROLE\"|ROLE|\"MY_PARENT_ROLE\""

**Reply by sfc-gh-asawicki** [reply]
Ugh, I just realized that the docs have errors: they don't name the resource that is imported. It should be:
```
terraform import snowflake_grant_account_role.example "\"MY_CHILD_ROLE\"|ROLE|\"MY_PARENT_ROLE\""
```

**Reply by sfc-gh-asawicki** [reply]
I fixed the docs: https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3183. Starting from v0.98.0 (which will be released later today), the docs on grant resources imports should be correct.

**Reply by jayjay-ayjay** [reply, marked as answer]
Info for other readers: 
sfc-gh-asawicki and I solved it as follows:
terraform import module.my_snowflake_module[\"MY_PARENT_ROLE\"].snowflake_grant_account_role.role_grants[\"MY_CHILD_ROLE\"] '"MY_CHILD_ROLE"|ROLE|"MY_PARENT_ROLE"'

My role functionality is inside a module. Otherwise the default statement should be used though.
